### PR TITLE
Fix default values for redirect list updates.

### PR DESCRIPTION
**Pull request by orium**

### CODE REVIEW

File: internal/provider/resource_cloudflare_list.go [hunk 1]
```diff
@@ -209,8 +209,8 @@ func buildListItemsCreateRequest(resource *schema.ResourceData, items []interfac
 			var preservePathSuffix *bool = nil
 
 			hasField := func(field string) bool {
-				_, has := resource.GetOkExists(fmt.Sprintf("item.%d.value.0.redirect.0.%s", i, field))
-				return has
+				v := getRawValue(fmt.Sprintf("item.%d.value.0.redirect.0.%s", i, field), resource.GetRawConfig())
```
Comment (jacobbednarz) [inline internal/provider/resource_cloudflare_list.go:212]: we really shouldn't be using `getRawValue` as it calls Terraform internal methods without going via the abstraction layers. it was initially planned to help with tristate booleans in Terraform configs but we instead took another approach due to the stability and on going issues with it. it really should have been removed.

what is the issue we're hitting here that requires it?

Comment (jacobbednarz) [reply internal/provider/resource_cloudflare_list.go:212]: this is the exact issue type we ended up solving in #1689 with a breaking change. as i suspect it is the same underlying cause of tristate boolean values, here was the testing outcome i did then with `d.GetRawConfig`.

|Initial state| Updated state|Outcome|
|---|---|---|
|enabled: false|enabled: true| ✅ |
| enabled: false | enabled: nil | ❌ |
| enabled: true | enabled: false | ✅ |
| enabled: true | enabled: nil | ✅ |
| enabled: nil | enabled: true | ✅ |
| enabled: nil | enabled: false | ❌ |

as you can see, this didn't fully solve the issue despite getting us most of the way there. another drawback is that this approach is using internal methods to Terraform and isn't recommended to be relied upon.

i'll grab some time with you to discuss what we can do going forward here.

Comment (orium) [reply internal/provider/resource_cloudflare_list.go]: :( It really shouldn't be this hard to get optional values to work.  In any case, I've updated the PR and avoided the issue by negating the field name, so the default is now `false`.  This is, of course, a breaking change, but the field was misbehaving any way, and was just introduced in the last release.

Comment (jacobbednarz) [reply internal/provider/resource_cloudflare_list.go]: agree; i've flicked you an internal ticket that covers the context and our potential next steps. as this is an issue in Terraform core (not the provider) we are pretty limited in the options to take. 

eventually, we'll move to terraform-plugin-framework which doesn't have this issue with uninitialised boolean values.

**Comment by jacobbednarz** [comment]
i'm having a look through this PR and would supplying `false` as the default solve the issue here? also, i don't think the latest commit actually solves this as it is still a boolean - for the workaround to work (heh), we'll likely need a string that allows "enabled"/"disabled"/"" as the values.

File: internal/provider/schema_cloudflare_list.go [hunk 2]
```diff
@@ -69,29 +69,33 @@ var listItemElem = &schema.Resource{
...
+									Description:  "Whether the redirect target url should keep the query string of the request's url.",
+									Type:         schema.TypeString,
+									Optional:     true,
+									ValidateFunc: validation.StringInSlice([]string{"disabled", "enabled"}, false),
 								},
 								"preserve_path_suffix": {
-									Description: "Whether to preserve the path suffix when doing subpath matching.",
-									Type:        schema.TypeBool,
-									Optional:    true,
+									Description:  "Whether to preserve the path suffix when doing subpath matching.",
```
Comment (jacobbednarz) [inline internal/provider/schema_cloudflare_list.go:95]:
```suggestion
									Description:  fmt.Sprintf("Whether to preserve the path suffix when doing subpath matching. %s", renderAvailableDocumentationValuesStringSlice([]string{"disabled", "enabled"})),
```

**Comment by jacobbednarz** [comment]
acceptance tests are all passing 

```
TF_ACC=1 go test $(go list ./...) -v -run "^TestAccCloudflareList_" -count 1 -parallel 1 -timeout 120m -parallel 1
?       github.com/cloudflare/terraform-provider-cloudflare     [no test files]
=== RUN   TestAccCloudflareList_Exists
--- PASS: TestAccCloudflareList_Exists (10.63s)
=== RUN   TestAccCloudflareList_UpdateDescription
--- PASS: TestAccCloudflareList_UpdateDescription (18.67s)
=== RUN   TestAccCloudflareList_Update
--- PASS: TestAccCloudflareList_Update (49.49s)
PASS
ok      github.com/cloudflare/terraform-provider-cloudflare/internal/provider   79.346s
?       github.com/cloudflare/terraform-provider-cloudflare/tools/cmd/changelog-check   [no test files]
?       github.com/cloudflare/terraform-provider-cloudflare/tools/cmd/maintainer-only-file-check        [no test files]
?       github.com/cloudflare/terraform-provider-cloudflare/tools/cmd/tf-log-check      [no test files]
?       github.com/cloudflare/terraform-provider-cloudflare/version     [no test files]
```

**Pull request merged**